curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/handle-a-fulfilled-promise-with-then.english.md

```diff
@@ -0,0 +1,78 @@
+---
+id: 5cdafbd72913098997531681
+title: Handle a Fulfilled Promise with then
+challengeType: 1
+---
+
+## Description
+<section id='description'>
+Promises are most useful when you have a process that takes an unknown amount of time in your code, something asynchronous. Often, a server request. When you make a server request it takes some amount of time, and after it completes you usually want to do something with the response from the server. This can be achieved by using the <code>then</code> method. The <code>then</code> method is executed immediately after your promise is fulfilled with <code>resolve</code>. Here’s an example:
+
+```js
+myPromise.then(result => {
+  // do something with the result.
+});
+```
+
+<code>result</code> comes from the argument given to the <code>resolve</code> method.
+</section>
+
+## Instructions
+<section id='instructions'>
+Use the <code>then</code> method on your promise. Log its argument to the console.
+</section>
+
+## Tests
+<section id='tests'>
+
+```yml
+tests:
+  - text: You should call the <code>then</code> method on the promise.
+    testString: assert(code.match(/(makeServerRequest|\))\.then\(\s*(\s*function\s*\(\s*result\s*\)\s*|\s*result\s*=>\s*|\s*\(\s*result\s*\)\s*=>\s*){[^}]*}\s*\)/g));
```

Comment:
I don't have an immediate code suggestion here, but if the text of the test is going to be "You should call the `then` method on the promise", then it should really only be checking if at a minimum `.then()` is used.  The current test focuses on the construction of the callback function.  Maybe another test should be added to test for the existence of using some kind of callback function.

Reply:
I can try and come up with something a little more specific here.

Reply:
LGTM